vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -187,13 +189,19 @@ protected void writeDateTime(BsonWriter writer, String name, Object value, Encod
   @Override
   protected Object readBinary(BsonReader reader, DecoderContext ctx) {
     final JsonObject result = new JsonObject();
-    result.put(BINARY_FIELD, reader.readBinaryData().getData());
+    BsonBinary tmpObj = reader.readBinaryData();
+    result.put(BINARY_FIELD, tmpObj.getData())
+        .put(TYPE_FIELD, tmpObj.getType());
     return result;
```

Comment:
Type is always defined as it use byte primitive and if developer will not specify which one to use it will put default one.